ui/src/components/build/create_build_config/BuildSection.js

```diff
@@ -0,0 +1,188 @@
+  BUILD_PLATFORM_IOS,
+  BUILD_TYPE_RELEASE,
+  BUILD_TYPE_DEBUG,
+  KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE,
+  KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_PASSWORD,
+  KEY_CR_BUILD_ANDROID_CREDENTIALS_NAME,
+  KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_ALIAS,
+  KEY_CR_BUILD_IOS_CREDENTIALS_NAME,
+  KEY_CR_BUILD_IOS_CREDENTIALS_DEVELOPER_PROFILE,
+  KEY_CR_BUILD_IOS_CREDENTIALS_PROFILE_PASSWORD,
+  KEY_HIDE_PLATDORM
+} from '../Constants';
+import FormDropdown from '../../common/FormDropdown';
+import KeyValueEditor from '../../common/KeyValueEditor';
+import { Checkbox } from 'patternfly-react/dist/js/components/Form';
+import UploadControl from '../../common/UploadControl';
+
+function onEnvVarsChange(configState, rows) {
+  configState.set(
+    KEY_CR_ENV_VARS,
+    rows.map(({ key, value }) => ({
+      name: key,
+      value
+    }))
+  );
+}
+
+export function renderBuildSection(component) {
+  const { configState, buildState } = component;
+  const { externalCredentials, [KEY_HIDE_PLATDORM]: hidePlatform } = component.state;
+  const { platform = BUILD_PLATFORM_ANDROID, buildType = BUILD_TYPE_DEBUG } = buildState.getOrEmpty();
+  return (
+    <div className="section">
+      <h3 className="with-divider">Build Configuration</h3>
+      {hidePlatform ? (
+        <React.Fragment />
+      ) : (
+        <FormGroup>
+          <ControlLabel>Platform</ControlLabel>
+          <FormDropdown
+            id="platform"
+            items={[BUILD_PLATFORM_ANDROID, BUILD_PLATFORM_IOS]}
+            titles={['Android', 'iOS']}
+            onSelect={active => buildState.set(KEY_CR_BUILD_PLATFORM, active)}
+            selected={platform}
+          />
+        </FormGroup>
+      )}
+      <FormGroup>
+        <ControlLabel>Build Type</ControlLabel>
+        <FormDropdown
+          id="buildtype"
+          items={[BUILD_TYPE_DEBUG, BUILD_TYPE_RELEASE]}
+          titles={['Debug', 'Release']}
+          onSelect={active => buildState.set(KEY_CR_BUILD_TYPE, active)}
+          selected={buildType}
+        />
+      </FormGroup>
+      {platform === BUILD_PLATFORM_ANDROID && buildType === BUILD_TYPE_RELEASE ? (
+        <FormGroup>
+          <Checkbox
+            checked={externalCredentials}
+            onChange={e => component.setState({ externalCredentials: e.target.value })}
+          >
+            Upload external credentials (Use this option if your Keystore is external to your source code)
+          </Checkbox>
+        </FormGroup>
+      ) : (
+        <React.Fragment />
+      )}
+      {platform === BUILD_PLATFORM_IOS ? renderIOSCredentials(component) : <React.Fragment />}
+      {externalCredentials ? renderAndroidCredentials(component) : <React.Fragment />}
+      <FormGroup>
+        <ControlLabel>Environment Variables</ControlLabel>
+        <KeyValueEditor onChange={rows => onEnvVarsChange(configState, rows)} />
+      </FormGroup>
+    </div>
+  );
+}
+
+function renderAndroidCredentials(component) {
```

Comment:
I think this really should be a React component on its own.

Reply:
made own component